pkg/util/request_validation.go

```diff
@@ -0,0 +1,33 @@
+package util
+
+import (
+	"errors"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// nolint: revive // Moved error out of pkg/webhook/admission; needs capitalization for backwards compat.
+var ErrOldObjectIsNil = errors.New("For admission webhooks registered for DELETE operations, please use Kubernetes v1.15.0+.")
+
+// SetObjectOnDelete enforces that we use at least K8s API v1.15.0+ on DELETE operations
+// and copies over the oldObject into the Object field for the given AdmissionRequest.
+func SetObjectOnDelete(req *admission.Request) error {
+	if req.AdmissionRequest.Operation == admissionv1.Delete {
+		// oldObject is the existing object.
+		// It is null for DELETE operations in API servers prior to v1.15.0.
+		// https://github.com/kubernetes/website/pull/14671
+		if req.AdmissionRequest.OldObject.Raw == nil {
+			return ErrOldObjectIsNil
```

Comment:
do we want to keep this old workaround? i think GK min version is 1.16 now

Reply:
We do have the minimum k8s version documented here https://open-policy-agent.github.io/gatekeeper/website/docs/install/#minimum-kubernetes-version. I'm ok with removing it if others do not have concerns.

Reply:
my first thought was: "yeah, let's rip this out". Then I wanted to make the argument on the other side.

I think keeping these code lines in particular can help `gator` users to follow the k8s expected behavior. Since they would be touching this "opaque" concept of an `AdmissionReview`, it's worth it imo to nudge people in the right direction. For instance, if we don't have these lines, a `gator` user may not pass in an `OldObject` on `DELETE` and get the wrong take away from their testing.

Reply:
That use case is persuasive to me. I'll merge at EOD if there are no objections. If we want to remove it, we can always do a quick follow-up.

Reply:
SGTM